aperture.go

```diff
@@ -307,15 +306,44 @@ func fileExists(name string) bool {
 
 // getConfig loads and parses the configuration file then checks it for valid
 // content.
-func getConfig(configFile string) (*Config, error) {
+func getConfig() (*Config, error) {
+	// Pre-parse command line flags to determine whether we've been pointed
+	// to a custom config file.
 	cfg := &Config{}
-	b, err := ioutil.ReadFile(configFile)
-	if err != nil {
+	if _, err := flags.Parse(cfg); err != nil {
 		return nil, err
 	}
-	err = yaml.Unmarshal(b, cfg)
-	if err != nil {
+
+	// If a custom config file is provided, we require that it exists.
+	var mustExist bool
```

Comment:
surely we always require that one or the other exists?

Reply:
Now that we allow command line flags, you could _theoretically_ specify everything on the command line. In that case, we don't want to fail if somebody gives us all the flags we need on the cli but the default location file doesn't exit. Whereas when you specifically set `--configfile`, I think it's fair to expect it to be there?